project-management/src/main/java/life/qbic/projectmanagement/domain/model/measurement/NGSSpecificMeasurementMetadata.java

```diff
@@ -0,0 +1,103 @@
+package life.qbic.projectmanagement.domain.model.measurement;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
+import java.util.Objects;
+import java.util.Optional;
+import life.qbic.projectmanagement.domain.model.sample.SampleId;
+
+/**
+ * <b><class short description - 1 Line!></b>
+ *
+ * <p><More detailed description - When to use, what it solves, etc.></p>
+ *
+ * @since <version tag>
+ */
+@Embeddable
+public class NGSSpecificMeasurementMetadata {
+
+  @Column(name = "sampleId")
+  private SampleId measuredSample;
+
+  @Column(name = "indexI5")
+  private String indexI5;
+
+  @Column(name = "indexI7")
+  private String indexI7;
+
+  @Column(name = "comment")
+  private String comment;
+
+  protected NGSSpecificMeasurementMetadata() {
+  }
+
+  private NGSSpecificMeasurementMetadata(SampleId measuredSample, String indexI5, String indexI7,
+      String comment) {
+    this.measuredSample = measuredSample;
+    this.indexI5 = indexI5;
+    this.indexI7 = indexI7;
+    this.comment = comment;
+  }
+
+  public static NGSSpecificMeasurementMetadata create(SampleId sampleId, String indexI5,
+      String indexI7, String comment) {
+    Objects.requireNonNull(sampleId);
+    Objects.requireNonNull(indexI5);
+    Objects.requireNonNull(indexI7);
+    Objects.requireNonNull(comment);
+    // According to the NGS experts, there can be
+    // - Double indexing i5 +i7
+    // - Single indexing i7
+    // - no indexing (if no multiplexing is done)
+    if (!indexI5.isBlank() && !indexI7.isBlank()) {
+      // Double indexing
+      return new NGSSpecificMeasurementMetadata(sampleId, indexI5, indexI7, comment);
+    }
+    if (!indexI7.isBlank()) {
+      // Single indexing
+      return new NGSSpecificMeasurementMetadata(sampleId, indexI5, indexI7, comment);
+    }
+    if (!indexI5.isBlank()) {
```

Comment:
There is temporal coupling here. I suggest adding a check that i7 is not present. Otherwise, if the first condition is removed at some point, you might throw an erronous exception. You can also get rid of the first check by combining this check with `indexI7.isBlank()`

Reply:
The way you did it in the accessor `index()` is way easier to read.

Reply:
I think the issue here is that we want to ensure that we don't create an NGS Measurement with only an I5 index. To do it as we did it in the accessor, we would have to adapt the database and the entity to expect null values for the indices.